packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -454,12 +456,22 @@ def reconstruct(self, delete_obj: bool = True) -> np.ndarray:
             raise ValueError(f"Type for ring size {self.ring_size} was not found!")
 
         for share in self.child:
-            if not share.exists:
-                raise Exception(
-                    "One of the shares doesn't exist. This probably means the SMPC "
-                    "computation isn't yet complete. Try again in a moment or call .block.reconstruct()"
-                    "instead to block until the SMPC operation is complete which creates this variable."
-                )
+            # Check if share pointer is in processing stage
+            # If so, wait until it exists.
+            if share.client.processing_pointers.get(share.id_at_location, False):
```

Comment:
@rasswanth-s Can you check this? It could be that the old code was relying on the exception and a retry mechanism elsewhere in a green thread? Do you know if doing this change and using normal sleep here will cause any issues?

Reply:
Since the MPCTensor is run on the server side, this should work, but I think we could remove the unbounded wait using time. sleep(), as sometimes the smpc computation is not complete we would go in an infinite loop.

Reply:
Thanks for checking, yes that makes sense. I have added a timeout just like the above pointer code as well.